lib/engine/step/special_track.rb

```diff
@@ -70,41 +93,50 @@ def available_hex(entity, hex)
       end
 
       def hex_neighbors(entity, hex)
-        return unless (ability = tile_lay_abilities(entity))
-        return if ability.hexes&.any? && !ability.hexes&.include?(hex.id)
+        return unless (ability = abilities(entity))
+        return if !ability.hexes&.empty? && !ability.hexes&.include?(hex.id)
 
         operator = entity.owner.corporation? ? entity.owner : @game.current_entity
-        return if ability.reachable && !@game.graph.connected_hexes(operator)[hex]
+        return if ability.type == :tile_lay && ability.reachable && !@game.graph.connected_hexes(operator)[hex]
 
         @game.hex_by_id(hex.id).neighbors.keys
       end
 
       def potential_tiles(entity, hex)
-        return [] unless (tile_ability = tile_lay_abilities(entity))
+        return [] unless (tile_ability = abilities(entity))
 
         tiles = tile_ability.tiles.map { |name| @game.tiles.find { |t| t.name == name } }
         tiles = @game.tiles.uniq(&:name) if tile_ability.tiles.empty?
 
+        special = tile_ability.special if tile_ability.type == :tile_lay
         tiles
           .compact
-          .select { |t| @game.phase.tiles.include?(t.color) && @game.upgrades_to?(hex.tile, t, tile_ability.special) }
+          .select { |t| @game.phase.tiles.include?(t.color) && @game.upgrades_to?(hex.tile, t, special) }
       end
 
-      def tile_lay_abilities(entity, **kwargs, &block)
+      def abilities(entity, **kwargs, &block)
         return unless entity&.company?
 
         if @round.respond_to?(:just_sold_company) && entity == @round.just_sold_company
           ability = @game.abilities(entity, :tile_lay, time: 'sold', **kwargs, &block)
+          return ability if ability
+        end
+
+        %i[tile_lay teleport].each do |type|
+          ability = @game.abilities(entity,
+                                    type,
+                                    time: %w[special_track %current_step% owning_corp_or_turn],
+                                    **kwargs,
+                                    &block)
+          return ability if ability && (ability.type != :teleport || !ability.used?)
         end
 
-        ability ||
-          @game.abilities(
-            entity, :tile_lay, time: %w[special_track %current_step% owning_corp_or_turn], **kwargs, &block
-          )
+        nil
       end
 
       def check_connect(_action, ability)
         hex_ids = ability.hexes
+        return if ability.type == :teleport
```

Comment:
is this method only called when game is not loading?

Reply:
Looks like it. I added "unless @game.loading" where it is called.

Reply:
it might be safer if this method always returns when @Game.loading then you don't put the onus on the caller

Reply:
Agree. I didn't look closely enough to see all the method did was throw the game error. Fixed.